experiments/mountain_car_plots.py

```diff
@@ -0,0 +1,269 @@
+) -> None:
+  """Save mountain car reward heatmaps for every action and every checkpoint.
+
+  Plots with rollout transitions scatterplotted on top are saved in
+  "{output_dir}/act_{i}".
+  Plots without rollout transitions are saved in
+  "{output_dir}/no_rollout_act_{i}".
+
+  Args:
+      checkpoints_dir: Path to `checkpoint` directory from AIRL or GAIL output
+          directory. Should contain "gen_policy" and "discrim" directories.
+      output_dir: Heatmap output directory.
+      exp_trajs: Expert trajectories for scatterplotting.
+          Generator trajectories
+          are dynamically generated from generator checkpoints.
+  """
+  checkpoints_dir = Path(checkpoints_dir)
+  for checkpoint_dir in sorted(checkpoints_dir.iterdir()):
+    vec_normalize_path = checkpoint_dir / "gen_policy" / "vec_normalize.pkl"
+    discrim_path = checkpoint_dir / "discrim"
+    policy_path = checkpoint_dir / "gen_policy"
+
+    with open(vec_normalize_path, "rb") as f:
+      vec_normalize = pickle.load(f)  # type: VecNormalize
+    vec_normalize.training = False
+
+    gen_trajs = make_policy_rollout(policy_path)
+
+    reward_fn_ctx = load_reward("DiscrimNet", discrim_path, venv="unused")
+    with reward_fn_ctx as reward_fn:
+      norm_rew_fn = build_norm_reward_fn(reward_fn=reward_fn,
+                                         vec_normalize=vec_normalize)
+      for i in range(3):
+        make_heatmap(act=i, reward_fn=norm_rew_fn, gen_trajs=gen_trajs,
+                     exp_trajs=exp_trajs)
+        save_path = Path(output_dir, f"act_{i}", checkpoint_dir.name)
+        os.makedirs(save_path.parent, exist_ok=True)
+        plt.savefig(save_path)
+
+        make_heatmap(act=i, reward_fn=norm_rew_fn)
+        save_path = Path(output_dir, f"no_rollout_act_{i}",
+                         checkpoint_dir.name)
+        os.makedirs(save_path.parent, exist_ok=True)
+        plt.savefig(save_path)
+        print(f"saved to {save_path}")
+
+
+def plot_reward_vs_time(
+    trajs_dict: Dict[str, List[util.rollout.Trajectory]],
+    reward_fn: RewardFn,
+    colors: Sequence[str] = ("tab:blue", "tab:orange"),
+):
+  """Plots rewards received by many trajectories from many agents over time.
+
+  Args:
+      trajs_dict: Dictionary mapping rollout labels (e.g. "expert" or
+        "gen policy") to rollouts associated with those labels.
+      reward_fn: Reward function for evaluating rollout rewards.
+      colors: Custom colors for plotted rewards.
+  """
+  plt.clf()
+  for i, (traj_name, traj_list) in enumerate(trajs_dict.items()):
+    X = []
+    Y = []
+    for traj in traj_list:
+      T = len(traj.rews)
+      X.extend(range(T))
+      rews = reward_fn(traj.obs[:-1], traj.acts, traj.obs[1:], None)
+      Y.extend(rews)
+    if i < len(colors):
+      color = colors[i]
+    else:
+      color = None
+    plt.plot(X, Y, alpha=1.0, c=color, label=traj_name)
+  plt.xlabel("timestep")
+  plt.ylabel("test reward")
+  plt.legend()
+  plt.show()
```

Comment:
```suggestion
  return fig
```

Do not assume what the callee wants to do. Maybe they want to save it, maybe display it to a user, ...